packages/edit-site/src/components/page-patterns/use-patterns.js

```diff
@@ -184,29 +184,38 @@ const selectPatterns = createSelector(
 	]
 );
 
-const patternBlockToPattern = ( patternBlock, categories ) => ( {
-	blocks: parse( patternBlock.content.raw, {
+/**
+ * Converts a post of type `wp_block` to a 'pattern item' that more closely
+ * matches the structure of theme provided patterns.
+ *
+ * @param {Object} patternPost The `wp_block` record being normalized.
+ * @param {Map}    categories  A Map of user created categories.
+ *
+ * @return {Object} The normalized item.
+ */
+const convertPatternPostToItem = ( patternPost, categories ) => ( {
+	blocks: parse( patternPost.content.raw, {
 		__unstableSkipMigrationLogs: true,
 	} ),
-	...( patternBlock.wp_pattern_category.length > 0 && {
-		categories: patternBlock.wp_pattern_category.map(
+	...( patternPost.wp_pattern_category.length > 0 && {
+		categories: patternPost.wp_pattern_category.map(
 			( patternCategoryId ) =>
 				categories && categories.get( patternCategoryId )
 					? categories.get( patternCategoryId ).slug
 					: patternCategoryId
 		),
 	} ),
-	termLabels: patternBlock.wp_pattern_category.map( ( patternCategoryId ) =>
+	termLabels: patternPost.wp_pattern_category.map( ( patternCategoryId ) =>
 		categories?.get( patternCategoryId )
 			? categories.get( patternCategoryId ).label
 			: patternCategoryId
 	),
-	id: patternBlock.id,
-	name: patternBlock.slug,
-	syncStatus: patternBlock.wp_pattern_sync_status || PATTERN_SYNC_TYPES.full,
-	title: patternBlock.title.raw,
-	type: PATTERN_TYPES.user,
-	patternBlock,
+	id: patternPost.id,
+	name: patternPost.slug,
+	syncStatus: patternPost.wp_pattern_sync_status || PATTERN_SYNC_TYPES.full,
+	title: patternPost.title.raw,
+	type: patternPost.type,
```

Comment:
Here I've decided to pull the type from the post objects `type` property (which should be `wp_block`).

Given the discovery in https://github.com/WordPress/gutenberg/pull/57532#discussion_r1442452911 that there are two different `PATTERN_TYPE` objects that are inconsistent, using it here seems dangerous.